lib/auth/saml.go

```diff
@@ -352,12 +352,16 @@ func (a *Server) validateSAMLResponse(samlResponse string) (*samlAuthResponse, e
 		return nil, trace.AccessDenied(
 			"received response with incorrect or missing attribute statements, please check the identity provider configuration to make sure that mappings for claims/attribute statements are set up correctly. <See: https://goteleport.com/teleport/docs/enterprise/sso/ssh-sso/>, failed to retrieve SAML assertion info from response: %v.", err)
 	}
+	warningInfo, err := provider.VerifyAssertionConditions(&assertionInfo.Assertions[0])
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
```

Comment:
`provider.RetrieveAssertionInfo` calls `VerifyAssertionConditions` internally: https://github.com/russellhaering/gosaml2/blob/v0.6.0/retrieve_assertion.go#L67
You can revert all changes in this file.